src/Api/Request/AbstractRequest.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+
+namespace Silktide\ProspectClient\Api\Request;
+
+
+use Silktide\ProspectClient\HTTP\HttpWrapper;
+
+abstract class AbstractRequest
+{
+    protected $httpWrapper;
+
+    public function __construct(HttpWrapper $httpWrapper)
+    {
+        $this->httpWrapper = $httpWrapper;
```

Comment:
Let it be known I think this is icky - having a model with a dependency. Although as discussed the alternative also has its disadvantages

Reply:
Let it be known that I think you are icky!